src/comment.rs

```diff
@@ -211,3 +197,156 @@ fn comment_end() {
     assert_eq!(None, find_comment_end("// hi /* test */"));
     assert_eq!(Some(9), find_comment_end("// hi /*\n."));
 }
+
+
+/// Returns true if text contains any comment.
+pub fn contains_comment(text: &str) -> bool {
+    CodeClassedIterator::new(text.chars()).any(|(kind, _)| kind == CodeCharKind::Comment )
+}
+
+pub fn uncommented(text: &str) -> String {
+    CodeClassedIterator::new(text.chars()).filter_map(|(s, c)| match s { CodeCharKind::Normal => Some(c), CodeCharKind::Comment => None }).collect()
+}
+
+#[test]
+fn test_uncommented() {
+    assert_eq!(&uncommented("abc/*...*/"), "abc");
+    assert_eq!(&uncommented("// .... /* \n../* /* *** / */ */a/* // */c\n"), "..ac\n");
+    assert_eq!(&uncommented("abc \" /* */\" qsdf"), "abc \" /* */\" qsdf");
+}
+
+#[test]
+fn test_contains_comment() {
+    assert_eq!(contains_comment("abc"), false);
+    assert_eq!(contains_comment("abc // qsdf"), true);
+    assert_eq!(contains_comment("abc /* kqsdf"), true);
+    assert_eq!(contains_comment("abc \" /* */\" qsdf"), false);
+}
+
+struct CodeClassedIterator<T>
+    where T: Iterator,
+          T::Item: RichChar
+{
+    base: iter::Peekable<T>,
+    status: CodeClassedStatus,
+}
+
+trait RichChar {
+    fn get_char(&self) -> char;
+}
+
+impl RichChar for char {
+    fn get_char(&self) -> char {
+        return *self;
+    }
+}
+
+impl RichChar for (usize, char) {
+    fn get_char(&self) -> char {
+        return self.1;
+    }
+}
+
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+enum CodeClassedStatus {
+    Normal,
+    LitString,
+    LitStringEscape,
+    LitChar,
+    LitCharEscape,
+    BlockComment(u32),
```

Comment:
Could you add a comment explaining that the `u32` represents the comment nesting level?